autogen.sh

```diff
@@ -3,4 +3,6 @@
 libtoolize --force --copy &&
 autoreconf --force --install -I m4
 rm -Rf autom4te.cache
+sudo mkdir -p /var/run/redis/sonic-db/
+sudo cp ./tests/database_config.json /var/run/redis/sonic-db/
 
```

Comment:
Who is using it? This file is for build, not a good place.

Reply:
the test cases under sairedis/tests/ ,  sairedis/syncd/tests.cpp and sairedis/vslib/src/tests.cpp need this file , otherwise replaced new API are not runnable.
Those tests are triggered while building deb pkg, it is part of dh build I think it is ok to place it in file for building.  It run some testing like this:

make  check-TESTS
make[3]: Entering directory '/var/johnar/workspace/vs/sonic-sairedis-build-pr/sairedis/tests'
[92mRunning Aspell[0m
[92mWords to check: 2212[0m
@(#) International Ispell Version 3.1.20 (but really Aspell 0.60.7-20110707)
# KeyOpFieldsValuesTuple 1547
# UpdateObjectStatus 3148
& WRED 9 3288: RED, WIRED, WEED, WED, FRED, BRED, CRED, WREN, WE'D
# asicGetOperations 3855
# currentBestMatch 5685
# getNonRemovableObjects 7729
# getOidListFromAttribute 7752
# initViewRemovedVidSet 8562
# isNonRemovable 8947
# nonObjectIdMap 10280
# performObjectSetTransition 10989
# processAttributesForOids 11563
# processObjectForViewTransition 11588
# processSingleVid 11619
# redisGetAsicView 12300
PASS: aspellcheck.pl
Using scripts dir 'mlnx'
main::test_mlnx_nhg_member: [94mFresh start[0m
[94mKilling syncd[0m
[94mFlushing redis[0m
[94mStarting syncd[0m
[94mReplay full_nhg_member.rec[0m
[94mReplay empty_sw.rec[0m
main::test_mlnx_full_to_empty: [94mFresh start[0m
[94mKilling syncd[0m
[94mFlushing redis[0m
[94mStarting syncd[0m
[94mReplay full_no_hostif_entry.rec[0m
[94mReplay empty_sw.rec[0m
main::test_mlnx_empty_to_full_to_empty: [94mFresh start[0m

Reply:
If it is only for unit test, could you please use SonicDBConfig::initialize(file) instead of contaminate the build filesystem.

---
In reply to: [349826656](https://github.com/Azure/sonic-sairedis/pull/537#discussion_r349826656) [](ancestors = 349826656)

Reply:
for the tests.cpp, we can do that in each main(), I tried before, it works. But for the sairedis/tests/XXX, it is not CPP files, it is perl script which is using C++ compiled binary directly and we cannot use  SonicDBConfig::initialize(file) any more.

Reply:
an example is : 
 print color('bright_blue') . "Replay $file" . color('reset') . "\n";
  my @ret = `../saiplayer/saiplayer -u "$DIR/$file"`;


it is using saiplayer binary, which is compiled from saiplayer.cpp

Reply:
i agree with qi. For saiplayer, can you add a cmdline option to take the db config file?

Reply:
i was thinking about SAI_KEY for passing this value when doing api_initialize so it could be read service table functions, but go ahead and add any way and i will change that later sine those 2 options don't collide

Reply:
removed this part since it is handled via libswsscommon.